### PR TITLE
integrates rest of study flow / leaving Rally functionality into Core Add-On

### LINKED ISSUE #149: integrate study card flow into Core Add-On

### LINKED ISSUE #155: user expects cleanup on successfully "leaving study"
All joined studies must be uninstalled, deletion ping must be sent, and UX must accommodate somehow.

### PR DESCRIPTION
**Pull request by hamilton**
closes #149; closes #155

~Will tag others for review when this is ready.~

This PR is functionally complete and the UX is acceptable. Marking this ready for review for expediency's sake. I'm going to file some follow-up issues around small UX improvements & invisible code improvements.

This PR does touch a lot of files, but many of those changes are either very small or deletions (we're clearing out all the old remaining ion components in this PR).

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -37,19 +37,13 @@
   },
 
   "background": {
-    "scripts": [
-      "public/addon-build/background.js"
-    ]
+    "scripts": ["public/addon-build/background.js"]
```
Comment (Dexterp37) [inline manifest.json:40]: Why are we changing this file?

File: src/routes/Main.svelte [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            <PrivacyNotice />
+        {:else}
+            <CurrentStudies
+                sidebarOffset
+                studies={$store.availableStudies}
+                on:join-study={(evt) => { joinStudy(evt.detail); }}
+                on:leave-study={(evt) => { leaveStudy(evt.detail); }} />
+        {/if}
+</Layout>
+{/if}
```
Comment (Dexterp37) [inline src/routes/Main.svelte:45]: nit: missing newline

File: src/routes/current-studies/StudyCard.svelte [hunk 3]
```diff
@@ -1,3 +1,39 @@
+<script context=module>
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+ // ^^^ the linter currently forces this block even though it appears below.
```
Comment (Dexterp37) [inline src/routes/current-studies/StudyCard.svelte:5]: Do we have an issue about this? Can you file one if we don't?

**Review by Dexterp37 — APPROVED** [review]
This looks good to me with the nits below addressed.

File: manifest.json (hunk 1 above)
Comment (hamilton) [reply manifest.json:40]: hmm, looks like my linter did this for me.

File: src/routes/current-studies/StudyCard.svelte (hunk 3 above)
Comment (hamilton) [reply src/routes/current-studies/StudyCard.svelte:5]: yep, we have one filed.

File: manifest.json (hunk 1 above)
Comment (hamilton) [reply manifest.json]: Reverted. Turns out I can "save without autoformatting" in vscode!

**Pull request merged**